### PR TITLE
bulkExportReuse with POST and GET

### PR DESCRIPTION
**Pull request by LalithE**
**Describe the bug**
The reuse functionality does not return the same job Id for a bulk export task that has been requested with POST and then a GET export.

**Steps to Reproduce**
It shows up when somebody issues a POST/GET request for a bulk export job, and then uses a GET/POST to request the same job over again. Previously, it would have created a new job ID and not utilized the reuse function from bulk export. This fix remediates that.

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java [hunk 1]
```diff
@@ -220,6 +220,16 @@ public List<JobInstance> fetchInstances(FetchJobInstancesRequest theRequest, int
 		List<Batch2JobInstanceEntity> instanceEntities;
 
 		if (statuses != null && !statuses.isEmpty()) {
+			if (definitionId.equals("BULK_EXPORT")) {
```
Comment (davidraeside) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java:223]: Nitpick: maybe use a constant here

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java [hunk 2]
```diff
@@ -228,6 +238,30 @@ public List<JobInstance> fetchInstances(FetchJobInstancesRequest theRequest, int
...
+				if (objectNode.has("originalRequestUrl")) {
+					String url = objectNode.get("originalRequestUrl").asText();
+					if (url.contains("?")) {
+						objectNode.put("originalRequestUrl", url.split("\\?")[0]);
+					}
+				}
+				return mapper.writeValueAsString(objectNode);
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (davidraeside) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java:260]: definitely don't want to printStackTrace here... :)

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java [hunk 3]
```diff
@@ -228,6 +238,30 @@ public List<JobInstance> fetchInstances(FetchJobInstancesRequest theRequest, int
...
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			mapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
+			mapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+			JsonNode rootNode = mapper.readTree(theParams);
+
+			if (rootNode instanceof ObjectNode) {
+				ObjectNode objectNode = (ObjectNode) rootNode;
+
+				if (objectNode.has("originalRequestUrl")) {
```
Comment (steve-corbett-smilecdr) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java:251]: Nitpick: if you're repeating a String constant like `originalRequestUrl` three times in the same method, it should probably be in a variable to reduce the risk of typos.

**Review by steve-corbett-smilecdr — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java [hunk 4]
```diff
@@ -220,6 +225,11 @@ public List<JobInstance> fetchInstances(FetchJobInstancesRequest theRequest, int
 		List<Batch2JobInstanceEntity> instanceEntities;
 
 		if (statuses != null && !statuses.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (LalithE) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java:227]: Oh that was previous code not written by me

**Pull request merged**